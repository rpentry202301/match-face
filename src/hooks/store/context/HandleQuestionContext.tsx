"use client";

import React, { useContext } from "react";

type Refine = {
  word: string[];
  department: number[];
};

const SelectContext = React.createContext<[Refine, React.Dispatch<React.SetStateAction<Refine>>] | undefined>(undefined);

export function SelectProvider({ children }: { children: React.ReactNode }) {
  const [select, setSelect] = React.useState<Refine>({
    word: [],
    department: [],
  });

  return (
    <SelectContext.Provider value={[select, setSelect]}>
      {children}
    </SelectContext.Provider>
  );
}

export function useRefine() {
  const context = useContext(SelectContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a SelectProvider");
  }
  return context;
}
