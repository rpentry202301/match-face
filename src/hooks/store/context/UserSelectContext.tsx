"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { UsersReducer } from "../reducer/selecterReducer";
import { UsersReducerAction } from "@/types/admin/tasks/register/types";

export type Users = {
  id: string;
  name: string;
};

const UserSelectContext = createContext<
  [Users[], Dispatch<UsersReducerAction>] | undefined
>(undefined);

export const UserSelectProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<Users[], UsersReducerAction>>(
    // selecterReducer,
    UsersReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
