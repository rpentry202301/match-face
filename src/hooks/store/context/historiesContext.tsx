'use client'
import { createContext, Dispatch, useContext, useReducer } from "react";
import { SelectHistoryAction } from "@/types/admin/histories/admin_histories";
import { SelectHistoryReducer } from "../reducer/historiesReducer";

const SelectHistoryContext = createContext<{month:string,department:number[],skills:number[]}| undefined>(undefined);
const SelectHistoryDispatchContext = createContext<Dispatch<SelectHistoryAction>|undefined>(undefined)
const initialSelect = {month:"",department:[],skills:[]}

export function SelectHistoryProvider ({children}:{children:React.ReactNode}){
    const [state,dispatch] = useReducer(SelectHistoryReducer,initialSelect)
    return (
        <SelectHistoryContext.Provider value={state}>
            <SelectHistoryDispatchContext.Provider value={dispatch} >
            {children}
            </SelectHistoryDispatchContext.Provider>
        </SelectHistoryContext.Provider>
    )
}

export const useSelectHistory = () => {
    const context = useContext(SelectHistoryContext)
    if(context === undefined) {
        throw new Error("Function 'useSelectHistory' must be used in Provider.")
    }
    return context
}

export const useSelectHistoryDispatch = () => {
    const dispatch = useContext(SelectHistoryDispatchContext)
    if(dispatch === undefined) {
        throw new Error("Function 'useSelectHistoryDispatch' must be used in Provider.")
    }
    return dispatch
}
