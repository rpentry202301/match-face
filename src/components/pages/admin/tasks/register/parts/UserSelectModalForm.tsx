"use client";
import { useState, ReactNode, ChangeEvent, FormEvent } from "react";
import { createPortal } from "react-dom";
import WhiteButton from "@/components/ui/button/WhiteButton";
import WhiteButtonCheckBox from "@/components/pages/admin/tasks/register/parts/WhiteButtonCheckBox";
import OrangeButton from "@/components/ui/button/OrangeButton";
import SelectBox from "@/components/ui/selectbox/SelectBox";
import UserList from "./UserList";
import { Users, useUserSelect } from "@/hooks/store/context/UserSelectContext";
import type { FetchUserModalData } from "@/types/admin/tasks/register/types";

const UserSelectModalForm = ({
  fetchData,
}: {
  fetchData: FetchUserModalData;
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const [users, setUsers] = useState(fetchData.users);

  const initDepartments = fetchData.departments.map((dep) => {
    return {
      id: dep.id,
      label: dep.name,
      checked: false,
    };
  });

  const initState = fetchData.statuses.map((state) => {
    return {
      id: state.id,
      label: state.name,
      checked: false,
    };
  });

  const groupValues = fetchData.userGroups.map((group) => group.name);
  groupValues.unshift("");

  const [formData, setFormData] = useState({
    search: "",
    year: "",
    month: "",
    department: initDepartments,
    state: initState,
    group: "",
  });

  const [userSelect, userSelectDispatch] = useUserSelect();
  const [checkedValues, setCheckedValue] = useState<Users[]>(userSelect);

  const open = async () => {
    setCheckedValue(userSelect);
    setIsOpened(true);
  };

  const close = () => setIsOpened(false);

  // 2000年からの配列を作成
  const maxYear = new Date().getFullYear();
  const yearArr = Array(maxYear - 1999)
    .fill(2000)
    .map((num, index) => `${num + index}`);
  yearArr.unshift("");

  // 月の配列を作成
  const monthArr = Array(12)
    .fill(1)
    .map((num, index) => `${num + index}`);
  monthArr.unshift("");

  const handleChange = (
    e: ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  // チェックボックスは型が異なるため別の関数で
  const handleChangeCheckBox = (
    e: ChangeEvent<HTMLInputElement>,
    name: "department" | "state"
  ) => {
    const newData = formData[name];
    newData.map((data) => {
      if (data.label === e.target.value) {
        data.checked = !data.checked;
      }
      return newData;
    });
    setFormData({ ...formData, [name]: newData });
  };

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const name = formData.search;
    const hireDate =
      formData.year && formData.month
        ? `${formData.year}-${
            Number(formData.month) < 10
              ? `0${formData.month}`
              : `${formData.month}`
          }-01`
        : "";
    const checkedDepId = formData.department
      .filter((data) => data.checked)
      .map((dep) => dep.id);
    const checkedStateId = formData.state
      .filter((data) => data.checked)
      .map((state) => state.id);
    const selectedGroup = fetchData.userGroups.filter(
      (group) => group.name === formData.group
    );

    const searchQuerys = [
      name ? `name=${name}` : "",
      hireDate ? `hireDate=${hireDate}` : "",
      checkedDepId.length ? `departmentId=${checkedDepId}` : "",
      checkedStateId.length ? `statusId=${checkedStateId}` : "",
      selectedGroup?.length ? `groupId=${selectedGroup[0].id}` : "",
    ];

    const query = searchQuerys.filter((query) => query !== "").join("&");

    fetch(`/api/admin/tasks/register/users?${query}`, {
      cache: "no-cache",
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then((res) => {
        // console.log(res.status);
        return res.json();
      })
      .then((data) => {
        setUsers(data.userList);
      })
      .catch((err) => console.log(err));
  };

  const handleChangeUserList = (e: ChangeEvent<HTMLInputElement>) => {
    if (checkedValues.map((value) => value.name).includes(e.target.value)) {
      setCheckedValue(
        checkedValues.filter((value) => value.name !== e.target.value)
      );
    } else {
      setCheckedValue([
        ...checkedValues,
        { id: e.target.id, name: e.target.value },
      ]);
    }
  };

  const handleClose = () => {
    userSelectDispatch({ type: "select", payload: checkedValues });
    close();
  };

  return (
    <Modal buttonText="追加" isOpened={isOpened} open={open} close={close}>
      <div>
        <h2 className="text-base ml-3">▶️ユーザーを選択する</h2>
      </div>
      <form
        onSubmit={(e) => handleSubmit(e)}
        className="flex flex-col justify-center items-center border-2 w-10/12 px-10 py-5 mt-3 mx-auto"
      >
        <div className="flex flex-col gap-5 w-fit mx-auto">
          <div className="flex items-center">
            <input
              id="search"
              name="search"
              type="text"
              value={formData.search}
              onChange={handleChange}
              className="border-light-gray border-2 text-xs p-1 w-10/12"
              data-testid="search-box"
            />
            <WhiteButton label="検索" className="text-xs ml-4 w-16" />
          </div>
          <div className="flex items-center gap-1">
            <SelectBox
              optionVal={yearArr}
              value={formData.year}
              name="year"
              onChange={handleChange}
              className="text-xs p-1"
              id="year"
              data-testid="select-year"
            />
            <span className="text-xs">年</span>
            <SelectBox
              optionVal={monthArr}
              value={formData.month}
              name="month"
              onChange={handleChange}
              className="text-xs p-1"
              id="month"
              data-testid="select-month"
            />
            <span className="text-xs">月入社</span>
          </div>
          <div className="flex gap-4 flex-wrap">
            {fetchData.departments.map((element) => (
              <WhiteButtonCheckBox
                key={`dep_${element.id}`}
                id={`dep_${element.id}`}
                label={element.name}
                value={element.name}
                checked={
                  formData.department.filter(
                    (data) => data.label === element.name
                  )[0].checked
                }
                name="department"
                className="text-xs px-1 w-16"
                onChange={(e) => handleChangeCheckBox(e, "department")}
              />
            ))}
          </div>
          <div className="flex gap-4">
            {fetchData?.statuses.map((element) => {
              return element.name.length < 4 ? (
                <WhiteButtonCheckBox
                  key={`status_${element.id}`}
                  id={`status_${element.id}`}
                  label={element.name}
                  value={element.name}
                  checked={
                    formData.state.filter(
                      (data) => data.label === element.name
                    )[0].checked
                  }
                  name="state"
                  className="text-xs w-16 px-1 whitespace-nowrap"
                  onChange={(e) => handleChangeCheckBox(e, "state")}
                />
              ) : (
                <WhiteButtonCheckBox
                  key={`status_${element.id}`}
                  id={`status_${element.id}`}
                  label={element.name}
                  value={element.name}
                  checked={
                    formData.state.filter(
                      (data) => data.label === element.name
                    )[0].checked
                  }
                  name="state"
                  className="text-xs px-1 w-24"
                  onChange={(e) => handleChangeCheckBox(e, "state")}
                />
              );
            })}
          </div>
          <div className="flex flex-col gap-2">
            <div>
              <h2>▶️グループから検索する</h2>
            </div>
            <div>
              <SelectBox
                optionVal={groupValues}
                value={formData.group}
                name="group"
                onChange={handleChange}
                className="w-10/12 text-xs p-1"
                id="group"
                data-testid="select-group"
              />
            </div>
          </div>
        </div>
        <div className="mx-auto mt-7">
          <OrangeButton type="submit" label="絞り込み" className="text-xs" />
        </div>
      </form>

      <UserList
        users={users || []}
        checkedValues={checkedValues.map((value) => value.name)}
        onChange={handleChangeUserList}
      />

      <div className="mx-auto mt-8 w-fit">
        <OrangeButton
          label="選択完了"
          className="text-xs"
          onClick={handleClose}
          data-testid="search-button"
        />
      </div>
    </Modal>
  );
};

// Modalは親コンポーネント内ではなく外で定義
const Modal = ({
  children,
  buttonText,
  canCloseByClickingBackground = true,
  isOpened,
  open,
  close,
}: {
  children: ReactNode;
  buttonText: string;
  canCloseByClickingBackground?: boolean;
  isOpened: boolean;
  open: () => void;
  close: () => void;
}) => {
  if (!isOpened) {
    return (
      <WhiteButton label={buttonText} onClick={open} className="text-xs" />
    );
  }

  // レンダリングするDOMをbodyに固定するためPortalを使用
  const elmModal = (
    <div className="fixed top-0 left-0 z-30 flex items-center justify-center w-full h-full">
      <div className="relative z-20 lg:w-3/4 md:w-4/5 w-11/12 p-5 max-w-3xl bg-white">
        {children}
      </div>
      {canCloseByClickingBackground ? (
        <div
          className="absolute top-0 left-0 w-full h-full bg-black opacity-30"
          onClick={close}
        />
      ) : (
        <div className="absolute top-0 left-0 w-full h-full bg-black opacity-30" />
      )}
    </div>
  );
  return createPortal(elmModal, document.body);
};

export default UserSelectModalForm;
