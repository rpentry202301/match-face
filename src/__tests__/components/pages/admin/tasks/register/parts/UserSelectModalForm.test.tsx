import UserSelectModalForm from "@/components/pages/admin/tasks/register/parts/UserSelectModalForm";
import { UserSelectProvider } from "@/hooks/store/context/UserSelectContext";
import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// コンポーネント内では初期表示用のfetchは無いので、ダミーデータを渡しています
export const userSelectModalFetchedData = {
  departments: [
    {
      id: 1,
      name: 'Java',
    },
    {
      id: 2,
      name: 'PHP',
    },
    {
      id: 3,
      name: 'FR',
    },
    {
      id: 4,
      name: 'CL',
    },
    {
      id: 5,
      name: 'ML',
    },
    {
      id: 6,
      name: 'QA',
    }
  ],
  statuses: [
    {
      id: 1,
      name: '研修中',
    },
    {
      id: 2,
      name: '待機中',
    },
    {
      id: 3,
      name: 'アサイン中',
    }
  ],
  userGroups: [
    {
      id: 1,
      name: 'グループ1',
    },
    {
      id: 2,
      name: 'グループ2',
    }
  ],
  users: [
    {
      id: 3,
      name: 'テスト珠子',
      hireDate: '2023-04-01',
      departmentId: 3,
      department: {
        id: 3,
        name: 'FR'
      },
      statusId: 3,
      status: {
        id: 3,
        name: 'アサイン中'
      },
    },
    {
      id: 4,
      name: 'テスト洋子',
      hireDate: '2023-04-01',
      departmentId: 4,
      department: {
        id: 4,
        name: 'CL'
      },
      statusId: 1,
      status: {
        id: 1,
        name: '研修中'
      },
    },
    {
      id: 1,
      name: 'テスト太郎',
      hireDate: '2023-01-01',
      departmentId: 1,
      department: {
        id: 1,
        name: 'Java'
      },
      statusId: 2,
      status: {
        id: 2,
        name: '待機中'
      },
    },
    {
      id: 2,
      name: 'テスト次郎',
      hireDate: '2023-01-01',
      departmentId: 2,
      department: {
        id: 2,
        name: 'PHP'
      },
      statusId: 1,
      status: {
        id: 1,
        name: '研修中'
      },
    }
  ]
}

// モーダル展開用関数
const open = () => {
  render(
    <UserSelectProvider>
      <UserSelectModalForm fetchData={userSelectModalFetchedData} />
    </UserSelectProvider>
  );
  fireEvent.click(screen.getByRole("button"));
}

describe("UserSelectModalForm.tsx", () => {
  beforeEach(() => {
    // モーダルを開く
    open();
  })

  describe("スナップショットテスト", () => {
    it("モーダルを開いている状態", () => {
      const { baseElement, rerender } = render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      rerender(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      expect(baseElement).toMatchSnapshot();
    });
  })

  describe("入力テスト", () => {
    const user = userEvent.setup()
    it("検索ボックスの入力テスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      const input = screen.getByTestId("search-box") as HTMLInputElement;
      fireEvent.change(input, { target: { value: "田中太郎" } });
      expect(input.value).toBe("田中太郎");
    });
  
    it("入社年の選択テスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      // デフォルト選択が空文字かどうか
      const beforeSelect = screen.getByTestId("year:") as HTMLOptionElement;
      expect(beforeSelect.selected).toBe(true);
      // 今年の入社年を選択してテスト
      const thisYear = new Date().getFullYear();
      await user.selectOptions(screen.getByTestId("select-year"), `${thisYear}`);
      const afterSelect = screen.getByTestId(`year:${thisYear}`) as HTMLOptionElement;
      expect(afterSelect.selected).toBe(true);
    });

    it("入社月の選択テスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      const beforeSelect = screen.getByTestId("month:") as HTMLOptionElement;
      expect(beforeSelect.selected).toBe(true);
      // 1月を選択してテスト
      await user.selectOptions(screen.getByTestId("select-month"), "1")
      const afterSelect = screen.getByTestId("month:1") as HTMLOptionElement;
      expect(afterSelect.selected).toBe(true);
    })

    it("部署のチェックボックステスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      const beforeCheck = screen.getByTestId("check-box:FR") as HTMLInputElement;
      expect(beforeCheck.checked).toBe(false);
      // Javaを選択してテスト
      await user.click(screen.getByTestId("check-box:FR"))
      const afterCheck = screen.getByTestId("check-box:FR") as HTMLInputElement;
      expect(afterCheck.checked).toBe(true);
    })

    it("ステータスのチェックボックステスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      const beforeCheck = screen.getByTestId("check-box:待機中") as HTMLInputElement;
      expect(beforeCheck.checked).toBe(false);
      // 研修中を選択してテスト
      await user.click(screen.getByTestId("check-box:待機中"))
      const afterCheck = screen.getByTestId("check-box:待機中") as HTMLInputElement;
      expect(afterCheck.checked).toBe(true);
    })

    it("ユーザーグループの選択テスト", async () => {
      render(
        <UserSelectProvider>
          <UserSelectModalForm fetchData={userSelectModalFetchedData} />
        </UserSelectProvider>
      );
      const nowGroup = "";
      const beforeSelect = screen.getByTestId(`group:${nowGroup}`) as HTMLOptionElement;

      expect(beforeSelect.selected).toBe(true);
      // 別のグループを選択
      const newGroup = userSelectModalFetchedData.userGroups[1].name;
      await user.selectOptions(screen.getByTestId("select-group"), newGroup);
      const afterSelect = screen.getByTestId(`group:${newGroup}`) as HTMLOptionElement;
      expect(afterSelect.selected).toBe(true);
    })
  });
});
