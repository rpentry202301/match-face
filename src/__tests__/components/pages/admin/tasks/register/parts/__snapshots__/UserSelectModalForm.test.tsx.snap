// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelectModalForm.tsx スナップショットテスト モーダルを開いている状態 1`] = `
<body>
  <div />
  <div
    class="fixed top-0 left-0 z-30 flex items-center justify-center w-full h-full"
  >
    <div
      class="relative z-20 lg:w-3/4 md:w-4/5 w-11/12 p-5 max-w-3xl bg-white"
    >
      <div>
        <h2
          class="text-base ml-3"
        >
          ▶️ユーザーを選択する
        </h2>
      </div>
      <form
        class="flex flex-col justify-center items-center border-2 w-10/12 px-10 py-5 mt-3 mx-auto"
      >
        <div
          class="flex flex-col gap-5 w-fit mx-auto"
        >
          <div
            class="flex items-center"
          >
            <input
              class="border-light-gray border-2 text-xs p-1 w-10/12"
              data-testid="search-box"
              id="search"
              name="search"
              type="text"
              value=""
            />
            <button
              class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs ml-4 w-16"
            >
              検索
            </button>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <select
              class="border-2 border-light-gray cursor-pointer text-xs p-1"
              data-testid="select-year"
              id="year"
              name="year"
            >
              <option
                data-testid="year:"
                value=""
              />
              <option
                data-testid="year:2000"
                value="2000"
              >
                2000
              </option>
              <option
                data-testid="year:2001"
                value="2001"
              >
                2001
              </option>
              <option
                data-testid="year:2002"
                value="2002"
              >
                2002
              </option>
              <option
                data-testid="year:2003"
                value="2003"
              >
                2003
              </option>
              <option
                data-testid="year:2004"
                value="2004"
              >
                2004
              </option>
              <option
                data-testid="year:2005"
                value="2005"
              >
                2005
              </option>
              <option
                data-testid="year:2006"
                value="2006"
              >
                2006
              </option>
              <option
                data-testid="year:2007"
                value="2007"
              >
                2007
              </option>
              <option
                data-testid="year:2008"
                value="2008"
              >
                2008
              </option>
              <option
                data-testid="year:2009"
                value="2009"
              >
                2009
              </option>
              <option
                data-testid="year:2010"
                value="2010"
              >
                2010
              </option>
              <option
                data-testid="year:2011"
                value="2011"
              >
                2011
              </option>
              <option
                data-testid="year:2012"
                value="2012"
              >
                2012
              </option>
              <option
                data-testid="year:2013"
                value="2013"
              >
                2013
              </option>
              <option
                data-testid="year:2014"
                value="2014"
              >
                2014
              </option>
              <option
                data-testid="year:2015"
                value="2015"
              >
                2015
              </option>
              <option
                data-testid="year:2016"
                value="2016"
              >
                2016
              </option>
              <option
                data-testid="year:2017"
                value="2017"
              >
                2017
              </option>
              <option
                data-testid="year:2018"
                value="2018"
              >
                2018
              </option>
              <option
                data-testid="year:2019"
                value="2019"
              >
                2019
              </option>
              <option
                data-testid="year:2020"
                value="2020"
              >
                2020
              </option>
              <option
                data-testid="year:2021"
                value="2021"
              >
                2021
              </option>
              <option
                data-testid="year:2022"
                value="2022"
              >
                2022
              </option>
              <option
                data-testid="year:2023"
                value="2023"
              >
                2023
              </option>
            </select>
            <span
              class="text-xs"
            >
              年
            </span>
            <select
              class="border-2 border-light-gray cursor-pointer text-xs p-1"
              data-testid="select-month"
              id="month"
              name="month"
            >
              <option
                data-testid="month:"
                value=""
              />
              <option
                data-testid="month:1"
                value="1"
              >
                1
              </option>
              <option
                data-testid="month:2"
                value="2"
              >
                2
              </option>
              <option
                data-testid="month:3"
                value="3"
              >
                3
              </option>
              <option
                data-testid="month:4"
                value="4"
              >
                4
              </option>
              <option
                data-testid="month:5"
                value="5"
              >
                5
              </option>
              <option
                data-testid="month:6"
                value="6"
              >
                6
              </option>
              <option
                data-testid="month:7"
                value="7"
              >
                7
              </option>
              <option
                data-testid="month:8"
                value="8"
              >
                8
              </option>
              <option
                data-testid="month:9"
                value="9"
              >
                9
              </option>
              <option
                data-testid="month:10"
                value="10"
              >
                10
              </option>
              <option
                data-testid="month:11"
                value="11"
              >
                11
              </option>
              <option
                data-testid="month:12"
                value="12"
              >
                12
              </option>
            </select>
            <span
              class="text-xs"
            >
              月入社
            </span>
          </div>
          <div
            class="flex gap-4 flex-wrap"
          >
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:Java"
                id="dep_1"
                name="department"
                type="checkbox"
                value="Java"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_1"
              >
                Java
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:PHP"
                id="dep_2"
                name="department"
                type="checkbox"
                value="PHP"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_2"
              >
                PHP
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:FR"
                id="dep_3"
                name="department"
                type="checkbox"
                value="FR"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_3"
              >
                FR
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:CL"
                id="dep_4"
                name="department"
                type="checkbox"
                value="CL"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_4"
              >
                CL
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:ML"
                id="dep_5"
                name="department"
                type="checkbox"
                value="ML"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_5"
              >
                ML
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:QA"
                id="dep_6"
                name="department"
                type="checkbox"
                value="QA"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_6"
              >
                QA
              </label>
            </div>
          </div>
          <div
            class="flex gap-4"
          >
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:研修中"
                id="status_1"
                name="state"
                type="checkbox"
                value="研修中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 px-1 whitespace-nowrap cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="status_1"
              >
                研修中
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:待機中"
                id="status_2"
                name="state"
                type="checkbox"
                value="待機中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 px-1 whitespace-nowrap cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="status_2"
              >
                待機中
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                data-testid="check-box:アサイン中"
                id="status_3"
                name="state"
                type="checkbox"
                value="アサイン中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-24 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="status_3"
              >
                アサイン中
              </label>
            </div>
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <div>
              <h2>
                ▶️グループから検索する
              </h2>
            </div>
            <div>
              <select
                class="border-2 border-light-gray cursor-pointer w-10/12 text-xs p-1"
                data-testid="select-group"
                id="group"
                name="group"
              >
                <option
                  data-testid="group:"
                  value=""
                />
                <option
                  data-testid="group:グループ1"
                  value="グループ1"
                >
                  グループ1
                </option>
                <option
                  data-testid="group:グループ2"
                  value="グループ2"
                >
                  グループ2
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="mx-auto mt-7"
        >
          <div>
            <button
              class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
              type="submit"
            >
              絞り込み
            </button>
          </div>
        </div>
      </form>
      <div
        class="my-10"
      >
        <table
          class="w-10/12 border-collapse border-2 border-deep-gray text-sm mx-auto"
        >
          <thead
            class="bg-light-gray"
          >
            <tr>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
              />
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
              >
                入社日
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-20"
              >
                職種
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-28"
              >
                状態
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-fit"
              >
                氏名
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="テスト珠子"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-04-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                FR
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                アサイン中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                テスト珠子
              </td>
            </tr>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="テスト洋子"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-04-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                CL
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                研修中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                テスト洋子
              </td>
            </tr>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="テスト太郎"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-01-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                Java
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                待機中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                テスト太郎
              </td>
            </tr>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="テスト次郎"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-01-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                PHP
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                研修中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                テスト次郎
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="mx-auto mt-8 w-fit"
      >
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
            data-testid="search-button"
          >
            選択完了
          </button>
        </div>
      </div>
    </div>
    <div
      class="absolute top-0 left-0 w-full h-full bg-black opacity-30"
    />
  </div>
  <div>
    <button
      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
    >
      追加
    </button>
  </div>
</body>
`;
